packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -497,225 +566,153 @@ export default class SegmentationStateManager {
+   * Retrieves the segmentation representations for a given viewport.
+   * @param viewportId - The ID of the viewport.
+   * @param specifier - The specifier for the segmentation representations.
+   * @returns The segmentation representations for the given viewport, or undefined if not found.
    */
-  setSegmentationRepresentationConfig(
-    segmentationRepresentationUID: string,
-    config: RepresentationConfig
-  ): void {
-    const _config = this._getRepresentationConfig(
-      segmentationRepresentationUID
-    );
-
-    if (!_config) {
+  getSegmentationRepresentations(
+    viewportId: string,
+    specifier: {
+      segmentationId?: string;
+      type?: SegmentationRepresentations;
+    } = {}
+  ): SegmentationRepresentation[] | [] {
+    const segmentationRepresentations = this.state.viewports[viewportId];
+
+    if (!segmentationRepresentations) {
       return;
     }
 
-    _config.allSegments = config;
-  }
-
-  /**
-   * Sets the configuration for per-segment settings of a segmentation representation.
-   *
-   * @param segmentationRepresentationUID - The unique identifier of the segmentation representation.
-   * @param config - The configuration for per-segment settings.
-   */
-  setPerSegmentConfig(
-    segmentationRepresentationUID: string,
-    config: RepresentationConfig
-  ): void {
-    const _config = this._getRepresentationConfig(
-      segmentationRepresentationUID
-    );
-
-    if (!_config) {
-      return;
+    // If no specifier is provided, return all entries
+    if (!specifier.type && !specifier.segmentationId) {
+      return segmentationRepresentations;
     }
 
-    _config.perSegment = config;
+    return segmentationRepresentations.filter((representation) => {
+      const typeMatch = specifier.type
+        ? representation.type === specifier.type
+        : true;
+      const idMatch = specifier.segmentationId
+        ? representation.segmentationId === specifier.segmentationId
+        : true;
+      return typeMatch && idMatch;
+    });
   }
 
   /**
-   * Returns the visibility of a segmentation representation in a specific viewport.
+   * Retrieves the visibility of a segmentation representation in a specific viewport.
    * @param viewportId - The ID of the viewport.
-   * @param segmentationRepresentationUID - The UID of the segmentation representation.
+   * @param segmentationId - The ID of the segmentation.
+   * @param type - The type of the segmentation representation.
    * @returns The visibility of the segmentation representation in the viewport.
    */
   getSegmentationRepresentationVisibility(
     viewportId: string,
-    segmentationRepresentationUID: string
+    segmentationId: string,
+    type: SegmentationRepresentations
   ): boolean {
-    const viewport = this.state.viewports[viewportId];
-    return viewport && viewport[segmentationRepresentationUID]?.visible;
+    const viewportRepresentations = this.getSegmentationRepresentations(
```

Comment:
It seems to be a common pattern to get the segmentation representation specifically, eg the first one which matches.  Adding a funciton getSegmentationRepresentation(viewportId, segmentationid, type) return undefined or the representation seems like it would simplify this code.